frontend/packages/vkt/src/components/clerkEnrollment/listing/UpdateEnrollmentStatusButton.tsx

```diff
@@ -0,0 +1,80 @@
+import { ChevronRight } from '@mui/icons-material';
+import { CustomButton, LoadingProgressIndicator } from 'shared/components';
+import { APIResponseStatus, Color, Variant } from 'shared/enums';
+
+import { useClerkTranslation } from 'configs/i18n';
+import { useAppDispatch, useAppSelector } from 'configs/redux';
+import { EnrollmentStatus } from 'enums/app';
+import { ClerkEnrollment } from 'interfaces/clerkExamEvent';
+import { updateClerkEnrollmentStatus } from 'redux/reducers/clerkExamEventOverview';
+import { clerkExamEventOverviewSelector } from 'redux/selectors/clerkExamEventOverview';
+
+export const UpdateEnrollmentStatusButton = ({
+  enrollment,
+}: {
+  enrollment: ClerkEnrollment;
+}) => {
+  const { t } = useClerkTranslation({
+    keyPrefix:
+      'vkt.component.clerkEnrollmentListing.row.updateEnrollmentStatus',
+  });
+  const { clerkEnrollmentUpdateStatus, examEvent } = useAppSelector(
+    clerkExamEventOverviewSelector
+  );
+  const dispatch = useAppDispatch();
+
+  if (
+    !examEvent ||
+    [EnrollmentStatus.PAID, EnrollmentStatus.CANCELED].includes(
```

Comment:
Oliskohan loogisempi jos tää tarkistus tehtäis `ClerkEnrollmentListingRow`:ssa ja ei rendata tätä button-elementtiä lainkaan jos paid tai canceled? Tai jos siirtäis tosiaan sinne, mutta muuttaa vielä tuon tarkistuksen siihen muotoon että rendataan jos EXPECTING_PAYMENT tai QUEUED.

Reply:
Ja jos haluaa tätä vielä viedä vähän pidemmälle, niin tää komponentti vois ehkä ottaa `ClerkEnrollmentStatusUpdate`:n parameterina. Näin ei olis kahdessa komponentissa tieto siitä, mitkä tilat sallittuja / miten nyt päätellään uusi status vanhan perusteella.

Reply:
Siirsin tuon render checkin tuohon ylemmälle tasolle `ClerkEnrollmentListingRow` ja rendataan vain jos EXPECTING_PAYMENT tai QUEUED.

Avaatko tuota toista kohtaa. Toi sallittu tila napin rendaukselle nyt Row -tasolla mutta kummassakin keississä vain yhdessä paikassa tuo tsekkaus.

Reply:
Koitin tätä pyöritellä vähän ja käytännössä jotain tän tapaista siis meinasin, että löytyisi `ClerkEnrollmentListingRow`:n alta:

```
  const getStatusChange = () => {
    let newStatus;

    if (enrollment.status === EnrollmentStatus.EXPECTING_PAYMENT) {
      newStatus = EnrollmentStatus.QUEUED;
    } else if (enrollment.status === EnrollmentStatus.QUEUED) {
      newStatus = EnrollmentStatus.EXPECTING_PAYMENT;
    }

    return (
      newStatus && {
        id: enrollment.id,
        version: enrollment.version,
        newStatus,
      }
    );
  };

  const possibleStatusChange = getStatusChange();
```
ja tuolla alempana olis
```
        <TableCell align="right">
          {possibleStatusChange && (
            <ChangeEnrollmentStatusButton statusChange={possibleStatusChange} />
          )}
        </TableCell>
```
Ei tuosta tosin järin siisti tullut, mutta tässä se kiva puoli että ChangeEnrollmentStatusButtonin ei tarvitse erityisesti tietää siitä, mitkä tilansiirtymät ovat sallituja kun ne ovat yhden komponentin alla. Tällöin tarttis kai myös tuo `getStatusText` metodi ChangeEnrollmentStatusButtonin alla muuttaa siten, että ei tarvitsisi tsekkailla, mikä status nyt sattuu olemaan kyseessä vaan tulis joko arvo propsina tai lokalisaatio suoraan statuksen nimestä.

Mut en nyt tiedä onko tällainen oikeasti wörttiä lähteä tekemään. Vois varmaan jättää nykyiseen malliin, mutta tällaista tosiaan mietin että sais vastuut yhden komponentin alle tuosta tilansiirtymien hallinnasta. Alun perinhän ne itse asiassa kyllä oli yhden komponentin (ChangeEnrollmentStatusButtonin) alla, mut se tuntui taas ainakin jollain tasolla väärältä että tehtiin ehdollinen rendaus ko. komponentin sisällölle statuksen mukaan.

Reply:
Kattelin itse asiassa tätä vähän lisää ja sain tän tehtyä niin jos pushaan alle jonkinmoisen version niin voidaan joko käyttää sitä tai muokata lisää.